src/webgl/p5.Framebuffer.js

```diff
@@ -0,0 +1,552 @@
+import p5 from '../core/main';
+import * as constants from '../core/constants';
+import { checkWebGLCapabilities } from './p5.Texture';
+
+class FramebufferCamera extends p5.Camera {
+  constructor(fbo) {
+    super(fbo.target._renderer);
+    this.fbo = fbo;
+  }
+
+  _computeCameraDefaultSettings() {
+    super._computeCameraDefaultSettings();
+    this.defaultAspectRatio = this.fbo.width / this.fbo.height;
+    this.defaultEyeZ =
+      this.fbo.height / 2.0 / Math.tan(this.defaultCameraFOV / 2.0);
+    this.defaultCameraNear = this.defaultEyeZ * 0.1;
+    this.defaultCameraFar = this.defaultEyeZ * 10;
+  }
+
+  resize() {
+    // If we're using the default camera, update the aspect ratio
+    if (this.cameraType === 'default') {
+      this._computeCameraDefaultSettings();
+      this._setDefaultCamera();
+    } else {
+      this.perspective(
+        this.cameraFOV,
+        this.fbo.width / this.fbo.height
+      );
+    }
+  }
+}
+
+p5.FramebufferCamera = FramebufferCamera;
+
+class FramebufferTexture {
+  constructor(framebuffer, property) {
+    this.framebuffer = framebuffer;
+    this.property = property;
+  }
+
+  get width() {
+    return this.framebuffer.width * this.framebuffer.density;
+  }
+
+  get height() {
+    return this.framebuffer.height * this.framebuffer.density;
+  }
+
+  rawTexture() {
+    return this.framebuffer[this.property];
+  }
+}
+
+p5.FramebufferTexture = FramebufferTexture;
+
+class Framebuffer {
+  /**
+   * @class p5.Framebuffer
+   * @constructor
+   * @param {p5.Graphics|p5} target A p5 global instance or p5.Graphics
+   * @param {Object} [settings] A settings object
+   */
+  constructor(target, settings = {}) {
+    this.target = target;
+    this.target._renderer.framebuffers.add(this);
+
+    this.format = settings.format || constants.UNSIGNED_BYTE;
+    this.channels = settings.channels || (
+      target._renderer._pInst._glAttributes.alpha
+        ? constants.RGBA
+        : constants.RGB
+    );
+    this.depthFormat = settings.depthFormat || (
+      this.format === constants.UNSIGNED_BYTE
+        ? constants.UNSIGNED_BYTE
+        : constants.FLOAT
+    );
+    this.antialias = (
+      settings.antialias === undefined
+        ? target._pInst._glAttributes.antialias
+        : settings.antialias
+    );
+    if (this.antialias && target.webglVersion !== constants.WEBGL2) {
+      console.warn('Antialiasing is unsupported in a WebGL 1 context');
+      this.antialias = false;
+    }
+    if (settings.width && settings.height) {
```

Comment:
What happens if someone sets a width but not a height, or vice versa? In your code here it looks like it will just ignore that setting, but should we consider defaulting to the same for both dimensions if only one is provided? Another options would be to set the dimension that's provided, but then use a default for the one that isn't. 

I'm a little unsure what someone might expect to happen in this case. Perhaps we could throw a warn about whatever behavior we decide on.

Reply:
Yeah, one should probably always supply both, so I think a warning if only one is supplied would be good!